staging/src/k8s.io/client-go/transport/websocket/roundtripper.go

```diff
@@ -111,6 +113,14 @@ func (rt *RoundTripper) RoundTrip(request *http.Request) (retResp *http.Response
 	wsConn, resp, err := dialer.DialContext(request.Context(), request.URL.String(), request.Header)
 	if err != nil {
 		if errors.Is(err, gwebsocket.ErrBadHandshake) {
+			// Enhance the error message with the http.Response data if possible.
+			if resp != nil {
+				errInfo := bytes.NewBufferString(resp.Status)
+				if errorBody, readErr := io.ReadAll(resp.Body); readErr == nil && len(errorBody) != 0 {
+					errInfo.Write(errorBody)
```

Comment:
the PR description mentioned needing to be at high verbosity to see this... where is verbosity factored into using this detailed data

Reply:
The verbosity check is in the `fallback_dialer.go`

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/portforward/fallback_dialer.go#L52-L55

I added to the PR description to explain this.